### DISCUSSION TITLE
sonoma locking up on lock screen, apparently due to SHOWFULLNAME in loginwindow payload

### DISCUSSION THREAD

**Discussion by bernstei** [usnistgov/macos_security#348] [Q&A] [answered]
Has anyone else run across the Sonoma lockscreen lockup, similar to discussions like https://community.jamf.com/t5/jamf-nation/sonoma-lock-screen-wont-take-correct-password/m-p/300488 ?  It either won't accept the known correct password, or if it does, it beachballs forever (more than 20 minutes anyway). Only a hard power cycle gets it out once that happens.

That thread being on a jamf board notwithstanding, it doesn't seem to be related to jamf. I got the idea from that one (or some other discussion) that it might be the loginwindow mobileconfig payload. Removing that payload entirely does seem to fix it, but so does keeping the other payload elements, but just removing `SHOWFULLNAME=true`, which is set by https://github.com/usnistgov/macos_security/blob/c52b7ff0e44d6a004be0f0c5a11d5aa9b0402f47/rules/system_settings/system_settings_loginwindow_prompt_username_password_enforce.yaml#L63

Has anyone else seen this, and/or have a fix that makes it possible to satisfy that rule?

**Comment by jmahlman** [comment]
Have not personally seen this, have you opened an Apple support case? From the comments in that thread it seems like a macOS bug.

**Comment by bernstei** [comment]
I agree that it looks like an Apple bug, but we don't have a support contract, and there's not a lot of clear information online. I just hoped that someone here might have run across it.  We also seem (so far as I can tell so far) to have multiple machines and it's not affecting all of them, despite supposedly identical configurations, but I'll take a closer look at one of them in the next few days.

**Reply by georgalis** [reply]
It is not necessary to have a support contract to submit support requests. I have two issues resolved after initial support expired. The pattern is engage in chat, then get escalated for a resource callback. Have serial numbers, os versions, phone numbers, update history, etc, handy, and use a second computer. My first issue was resolved several months later in an update cycle, in the second I was navigated through good debugging for resolution (although their heuristics were less than ideal for my issue). They won't admit to a bug or problem as the first case, but they did offer to escalate my issue to their internal developers.

In this case, if there is a beach ball, spinning wheel, there should also be a key sequence to record a trace of the event creating the lockout.

**Comment by bernstei** [comment]
Thanks, I'll see if I can get anywhere with that.

**Comment by GrahamWatts** [comment]
Release notes of 14.2 might be relevant: https://support.apple.com/en-us/HT213893
"The login password is correctly accepted at the Lock Screen when MDM has configured the login window to hide admin users."

**Reply by bernstei** [reply]
Ooh, very interesting. I'll have to check that out.  If that's it, it may be arguable that setting the one setting without the other is a bug, and maybe I just want to request that the rules in `macos_security` will do both automatically as part of the Sonoma remediations.

**Reply by bernstei** [reply]
Do you happen to know if there's a key to do that for all admin users, or do I need to add them individually to the `HiddenUsersList` ? (which would be mildly annoying, since I'd have to add that step to future creation of new admin accounts).

**Reply by bernstei** [reply]
Sadly, no luck.  I hid the existing admin users with the `HiddenUsersList`, and they are indeed hidden, but the behavior once I set `SHOWFULLNAME` (switch from list of users to username/password text boxes) is unchanged. It still rejects the correct password unless I hit `escape` and select the correct user, and then it accepts the password but beachballs forever. I guess I may try to reproduce it on a clean system and file a support request.

**Reply by bernstei** [reply]
Aha - I finally found the `HideAdminUsers` loginwindow payload key in https://developer.apple.com/documentation/devicemanagement/loginwindow, and that did indeed appear to fix it.  Yeay.    Thanks you @GrahamWatts 

I do think that if that key is required for `SHOWFULLNAMES` to not break the unlocking process, `macos_security` should always include both if it includes `SHOWFULLNAME` at all. Unless someone convinces me otherwise, I'll open an issue for that.

**Reply by bernstei** [reply]
I'm also going to test the newly released 14.3 for this behavior.

**Reply by bernstei** [reply, marked as answer]
Tentatively looks like it's fine in 14.3, so maybe there's no need for a change to the `macos_security` behavior.
